core/src/kura.rs

```diff
@@ -339,176 +337,164 @@ impl Kura {
     }
 }
 
-/// The interface for the **block store**, which is where [Kura],
-/// the block storage subsystem, stores its blocks.
-///
-/// The [`BlockStoreTrait`] defines and implements functionality used by Kura
-/// for every platform. The default implementation is `StdFileBlockStore`,
-/// which uses `std::fs`.
-pub trait BlockStoreTrait: Debug {
-    /// Read a series of block indices from the block index file and
-    /// attempt to fill all of `dest_buffer`.
-    ///
-    /// # Errors
-    /// IO Error.
-    fn read_block_indices(
-        &self,
-        start_block_height: u64,
-        dest_buffer: &mut [(u64, u64)],
-    ) -> Result<()>;
+// Marker trait for typestate. Not sealed, since while it
+// doesn't need to be implemented by a user, implementing it
+// shouldn't lead to problematic behaviour.
+#[doc(hidden)]
+pub trait Lock: core::ops::Deref<Target = PathBuf> {}
 
-    /// Write the index of a single block at the specified `block_height`.
-    /// If `block_height` is beyond the end of the index file, attempt to
-    /// extend the index file.
-    ///
-    /// # Errors
-    /// IO Error.
-    fn write_block_index(&mut self, block_height: u64, start: u64, length: u64) -> Result<()>;
-
-    /// Get the number of indices in the index file, which is calculated as the size of
-    /// the index file in bytes divided by 16.
-    ///
-    /// # Errors
-    /// IO Error.
-    ///
-    /// The most common reason this function fails is
-    /// that you did not call `create_files_if_they_do_not_exist`.
-    ///
-    /// Note that if there is an error, you can be quite sure all other
-    /// read and write operations will also fail.
-    fn read_index_count(&self) -> Result<u64>;
-
-    /// Change the size of the index file (the value returned by
-    /// `read_index_count`).
-    ///
-    /// # Errors
-    /// IO Error.
-    ///
-    /// The most common reason this function fails is
-    /// that you did not call `create_files_if_they_do_not_exist`.
-    ///
-    /// Note that if there is an error, you can be quite sure all other
-    /// read and write operations will also fail.
-    fn write_index_count(&mut self, new_count: u64) -> Result<()>;
+/// Marker struct for typestate, signifies that store is
+/// not protected by a lockfile we own and thus can be used
```

Comment:
```suggestion
/// Marker struct for typestate. The marker signifies that the store is
/// not protected by a lockfile we own and thus can be used
```

who are `we` in this context?

Reply:
> who are we in this context?

This `Kura` instance. I'll clarify it.